contracts/strategies/allocation/direct-grants/DirectGrantsStrategy.sol

```diff
@@ -0,0 +1,317 @@
+    /// @notice Initialize the strategy
+    /// @param _poolId The pool ID for this strategy
+    /// @param _data The data to initialize the strategy with
+    function initialize(uint256 _poolId, bytes memory _data) public override {
+        super.initialize(_poolId, _data);
+
+        (registryGating) = abi.decode(_data, (bool));
+
+        registry = allo.getRegistry();
+    }
+
+    /// ===============================
+    /// =========== Views =============
+    /// ===============================
+
+    // Note: what determines a valid allocator?
+    function isValidAllocator(address _allocator) public view returns (bool) {
+        return allo.isPoolManager(poolId, _allocator);
+    }
+
+    function getRecipientStatus(address _recipientId) public view returns (RecipientStatus) {
+        return recipients[_recipientId].status;
+    }
+
+    /// @notice Get the status of the milestone
+    /// @param _milestoneId Id of the milestone
+    function getMilestoneStatus(uint256 _milestoneId) public view returns (RecipientStatus) {
+        // make sure it exists
+        if (milestones.length < _milestoneId) {
+            revert INVALID_MILESTONE();
+        }
+
+        return milestones[_milestoneId].milestoneStatus;
+    }
+
+    /// @notice Get the payout summary data for recipient
+    /// @param _recipientIds The recipient IDs to get the payout summary for. Note: this will only ever be 1 recipient for this strategy.
+    /// @param _data The data to get the payout summary for
+    /// @param _sender The sender of the transaction
+    function getPayouts(address[] memory _recipientIds, bytes memory _data, address _sender)
+        external
+        view
+        returns (PayoutSummary[] memory)
+    {
+        // Note: do we wan the total here or the amount for the milestone?
+        // we could calculate the amount for each milestone here and create the payouts array
+        // I am thinking we want an array of milestones, and then the amount for each milestone
+        uint256 milestoneCount = milestones.length;
+        PayoutSummary[] memory payouts = new PayoutSummary[](milestoneCount);
+        for (uint256 i = 0; i < milestoneCount;) {
+            // todo: calculate the amount for each milestone
+
+            // set the amount for each milestone and convert to a PayoutSummary object
+            // Note: amountPercentage in the value is just temporary, we will calculate the amount for each milestone
+            payouts[i] = PayoutSummary(acceptedRecipientId, milestones[i].amountPercentage);
+
+            unchecked {
+                i++;
+            }
+        }
+
+        return payouts;
+    }
+
+    /// ===============================
+    /// ====== External/Custom ========
+    /// ===============================
+
+    /// @notice Set milestones
+    /// @param _milestones The milestones to be set
+    function setMilestones(Milestone[] memory _milestones) external onlyPoolManager(msg.sender) {
```

Comment:
would need recipientId as well 
This would need to be updated to make sure we don't allow setting of milestones for project where a payment has already been made